### PR TITLE
Don't expand events with REC.-ID but no master event

### LINKED ISSUE #323: KeyError on files with RECURRENCE-ID but no "master"
Only on certain events, `khal` crashes like this:

		Traceback (most recent call last):
			File "/usr/bin/khal", line 9, in <module>
				load_entry_point('khal==0.7.1.dev41+ng699b6c3.d20160108', 'console_scripts', 'khal')()
			File "/usr/lib/python3.5/site-packages/click/core.py", line 716, in __call__
				return self.main(*args, **kwargs)
			File "/usr/lib/python3.5/site-packages/click/core.py", line 696, in main
				rv = self.invoke(ctx)
			File "/usr/lib/python3.5/site-packages/click/core.py", line 1037, in invoke
				return Command.invoke(self, ctx)
			File "/usr/lib/python3.5/site-packages/click/core.py", line 889, in invoke
				return ctx.invoke(self.callback, **ctx.params)
			File "/usr/lib/python3.5/site-packages/click/core.py", line 534, in invoke
				return callback(*args, **kwargs)
			File "/usr/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
				return f(get_current_context(), *args, **kwargs)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/cli.py", line 227, in cli
				ctx.invoke(cli.commands[command])
			File "/usr/lib/python3.5/site-packages/click/core.py", line 534, in invoke
				return callback(*args, **kwargs)
			File "/usr/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
				return f(get_current_context(), *args, **kwargs)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/cli.py", line 255, in calendar
				full=full
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/controllers.py", line 148, in calendar
				locale=locale)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/calendar_display.py", line 184, in vertical_month
				multiple, color, highlight_event_days, locale)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/calendar_display.py", line 128, in str_week
				devents = list(collection.get_events_on(day))
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/khalendar/khalendar.py", line 317, in <genexpr>
				return (self._cover_event(event) for event in events)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/khalendar/backend.py", line 433, in get_floating
				yield self.construct_event(item, href, start, end, ref, etag, calendar, dtype)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/khalendar/backend.py", line 515, in construct_event
				ref=ref,
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/khalendar/event.py", line 143, in fromString
				return cls.fromVEvents(events, ref, **kwargs)
			File "/usr/lib/python3.5/site-packages/khal-0.7.1.dev41+ng699b6c3.d20160108-py3.5.egg/khal/khalendar/event.py", line 136, in fromVEvents
				instcls = cls._get_type_from_vDDD(vevents[ref]['DTSTART'])
		KeyError: '1451779200'

This happens only occasionally and only on google calendars synced with vdirsyncer.
The timestamp causing the KeyError represents Jan 3rd at 1 AM. On this date, said google calendar contains an all-day event which recurs daily for about a week, but it seems like it is only the occurence on Jan 3rd causing this problem. In its `.ics` file, the event looks like this:

    BEGIN:VEVENT
    DTSTART;VALUE=DATE:20160103
    DTEND;VALUE=DATE:20160104
    DTSTAMP:20160108T175109Z
    UID:7[...]@google.com
    RECURRENCE-ID;VALUE=DATE:20160103
    CREATED:20151216T103115Z
    DESCRIPTION:
    LAST-MODIFIED:20160102T142954Z
    LOCATION:
    SEQUENCE:1
    STATUS:CONFIRMED
    SUMMARY: [...]
    TRANSP:TRANSPARENT
    [...]
    END:VEVENT

In my sqlite database, where the contents of `vevents` come from, all rows usually seem to have `PROTO` stored in their `ref` column. There is only one single row where this is not the case:

    sqlite> select * from recs_float where ref is not 'PROTO';
    1451779200|1451865600|dde22800042d40ed9566e83b7f0b70ea.ics|1451779200|1451779200|0|cranio

I couldn't get into why this event has its `DTSTART` timestamp or whatever in its `ref` row, nor could I find any difference between the event I suspect to be the cause of this and other all-day-events that seem ok.

### PR DESCRIPTION
**Pull request by geier**
Until now we expanded *all* events with a RRULE or RDATE, even if only
VEVENTs (with the same UID) with specified RECURRENCE-ID parameters were
present. In this case, the expansion is nonsensical, because we do not
know the start (and end) datetime of the master (but only of the
exceptions).

This fixes #323.

### CODE REVIEW

**Comment by geier** [comment]
This isn't the cleanest solution, but it works.
@untitaker could you have a look? (I believe you know what's happening in expand since you recently refactored it)

File: khal/khalendar/aux.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 logger = log.logger
 
 
-def expand(vevent, href=''):
+def expand(vevent, href='', expand=True):
```
Comment (untitaker) [inline khal/khalendar/aux.py]: Strongly dislike this, I'd prefer if we could move the entire logic into the expand function.

**Comment by untitaker** [comment]
Even with the cleanest solution, I'd prefer not having any of those workarounds in khal, especially as it seems this is a bug in vdirsyncer (and nowhere else) anyway.

**Comment by geier** [comment]
@untitaker: did you get any (private) feedback on pimutils/vdirsyncer#391?

**Comment by untitaker** [comment]
No.

**Comment by untitaker** [comment]
Can we close this?

**Comment by geier** [comment]
yes

**Pull request closed without merging**